src/ORM/Query.php

```diff
@@ -223,19 +270,26 @@ public function select(
         ExpressionInterface|Table|Association|Closure|array|string|float|int $fields = [],
         bool $overwrite = false
     ) {
-        if ($fields instanceof Association) {
-            $fields = $fields->getTarget();
-        }
+        if ($fields instanceof Closure) {
+            $fields = $fields($this);
```

Comment:
All closures for various methods need to be run right away so that we can pass instance of `ORM\Query` else they would get instance of `Database\SelectQuery`.